Makefile

```diff
@@ -147,7 +147,7 @@ HOSTCFLAGS		+= $(WARNINGS) $(DEFINES) -iquote include/
 export AFLAGS CFLAGS USERCLFAGS HOSTCFLAGS
 
 # Default target
-all: flog criu lib crit
+all: flog criu lib crit amdgpu_plugin
```

Comment:
@adrianreber you mentioned in https://github.com/checkpoint-restore/criu/pull/1519#issuecomment-864825556:
> Also, as you mentioned that the plugin requires additional dependencies, the plugin should not be built automatically just by typing make. The plugin should be built only when explicitly requested.

What would be the best way of explicitly requesting for the plugin to build?

Reply:
I would say to not add it to `all:`. Just provide the Makefile target `amdgpu_plugin` which can then be built by running `make amdgpu_plugin`. This way no user not interested in the GPU plugin would need to install additional development packages. Same is true for distribution packages. If a distribution wants to ship a plugin they should explicitly build it.

Reply:
Agreed and I can just remove it from all: but before I do that, I'd like to highlight that the current plugin Makefile detects the dependencies and doesn't build/install the plugin if the dependencies are not met but let CRIU and other components proceed with the build. Having it this way also gives a chance to trigger automation specially ZDTM tests since we are adding new hooks to criu making some changes in the core criu that will be exercised only with amdgpu plugin at the moment but probably this plugin directory might see more device file support in future perhaps for other devices/vendors.

We'll probably need some changes to the CI to test with make amdgpu_plugin if I decouple it from all: target. Please advice.

Reply:
I've updated the PR with the feedback addressed.  Now the plugin will only be build when user supplies amdgpu_plugin option to make or install. Besides, it will also check the necessary dependencies on libdrm. The plugin itself will be a positive test case for the new hooks and the dummy plugin can act as sort of placeholder as it returns -ENOTSUP for the hooks and gives a chance to other potential implementations of those new hooks. This should not break the exisitng criu build and test infra. Will wait for the CI results when available.

Reply:
I will update this PR along with #1556 as this depends on it.